lib/session_encryptor.rb

```diff
@@ -1,18 +1,180 @@
 class SessionEncryptor
+  class SensitiveKeyError < StandardError; end
+
+  class SensitiveValueError < StandardError; end
+  NEW_CIPHERTEXT_HEADER = 'v2'
+  SENSITIVE_KEYS = [
+    'first_name', 'middle_name', 'last_name', 'address1', 'address2', 'city', 'state', 'zipcode',
+    'zip_code', 'dob', 'phone_number', 'phone', 'ssn', 'prev_address1', 'prev_address2',
+    'prev_city', 'prev_state', 'prev_zipcode', 'pii', 'pii_from_doc', 'password', 'personal_key',
+    'email', 'email_address', 'unconfirmed_phone'
+  ].to_set.freeze
+
+  # 'idv/doc_auth' and 'idv' are used during the proofing process and can contain PII
+  # personal keys are generated and stored in the session between requests, but are used
+  # to decrypt PII bundles, so we treat them similarly to the PII itself.
+  SENSITIVE_PATHS = [
+    ['warden.user.user.session', 'idv/doc_auth'],
+    ['warden.user.user.session', 'idv'],
+    ['warden.user.user.session', 'personal_key'],
+    ['warden.user.user.session', 'unconfirmed_phone'],
+    ['flash', 'flashes', 'personal_key'],
+    ['flash', 'flashes', 'email'],
+    ['email'],
+  ]
+
+  # rubocop:disable Layout/LineLength
+  SENSITIVE_REGEX = %r{FAKEY|MIDDLEFAKER|MCFAKERSON|1111111111111|GREAT FALLS|1938-10-06|2099-12-31|314-555-1212}
+  # rubocop:enable Layout/LineLength
```

Comment:
Is it worth doing something similar to what was done in `FakeAnalytics`, referencing the constant from which these values are generated?

https://github.com/18F/identity-idp/blob/0e8c7e136f6552d17f13a77700868d84b133c709/spec/support/fake_analytics.rb#L23-L31

Reply:
Thank you that's a great catch, I had forgotten we had them in a constant.

SessionEncryptor being in `lib/` means it can't reference things outside of `lib/` so I moved it into our constants file in e11e91946, but I'm not sure if that's the best place

Reply:
When I was using it for `spec/factories/profiles.rb` in #6229, I had felt it was kinda weird to reference the constant where it was at, so I like the idea of having a single constant of fake PII in a common location like that.